### PR TITLE
On-demand deprecated concept caching

### PR DESCRIPTION
**Pull request by jasonleinbach-wf**
#### Changes:

- Create json caches on-demand, when not present.
- Removed unnecessary arelle logging in the rule.

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

**Comment by brianmyers-wf** [comment]
+1

File: dqc_us_rules/dqc_us_0018.py [hunk 1]
```diff
@@ -181,7 +173,9 @@ def deprecated_facts_errors(val):
     :return: No explicit return, though error messages are created
     :rtype: None
     """
-    _load_cache(val)
+    cache_exists = _load_cache(val)
+    if not cache_exists:
+        _create_cache(val)
```
Comment (andrewperkins-wf) [inline dqc_us_rules/dqc_us_0018.py]: does this also load the cache up?

Comment (jasonleinbach-wf) [inline dqc_us_rules/dqc_us_0018.py]: Yes.  The cache is loaded into val.usgaapDeprecations, then dumped to a file.  _load_cache takes the file's content and saves it under val.usgaapDeprecations.

**Comment by andrewperkins-wf** [comment]
+1 review

**Comment by andrewperkins-wf** [comment]
QA +10

# Manual Review
Fired DQC 18 rule as expected

# Unit Tests
```
.............................................................................
----------------------------------------------------------------------
Ran 77 tests in 0.043s
```

Flake8 issues are being worked on as a followup, and will be coming shortly.

**Pull request merged**